File: python/lists/max_min.py
# Find Maximum and Minimum in One Pass without using in-built functions

# Input list
numbers = [3, 1, 9, 4]

# Initialize max and min with the first element
maximum = numbers[0]
minimum = numbers[0]

# Traverse the list once
for num in numbers:
    # Update maximum if current number is greater
    if num > maximum:
        maximum = num

    # Update minimum if current number is smaller
    if num < minimum:
        minimum = num

# Print the results
print("Max =", maximum)
print("Min =", minimum)

# Output:
# Max = 9
# Min = 1

File: C/max_min.c
#include <stdio.h>

void findMinMax(int arr[], int n, int *min, int *max)
{
    int i;
    *min = *max = arr[0];

    for (i = 1; i < n; i++)
    {
        if (arr[i] > *max)
            *max = arr[i];
        else if (arr[i] < *min)
            *min = arr[i];
    }
}

int main()
{
    int n, i, min, max;

    printf("Enter the number of elements: ");
    scanf("%d", &n);

    if (n <= 0)
    {
        printf("Invalid array size.\n");
        return 0;
    }

    int arr[n];

    printf("Enter %d elements:\n", n);
    for (i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    findMinMax(arr, n, &min, &max);

    printf("Maximum element = %d\n", max);
    printf("Minimum element = %d\n", min);

    return 0;
}
